source/common/upstream/original_dst_cluster.cc

```diff
@@ -40,6 +41,26 @@ OriginalDstCluster::LoadBalancer::LoadBalancer(PrioritySet& priority_set, Cluste
 
 HostConstSharedPtr OriginalDstCluster::LoadBalancer::chooseHost(LoadBalancerContext* context) {
   if (context) {
+    // Check if request has a override host header set, if directly return the host using it.
+    const Http::HeaderMap* downstream_headers = context->downStreamHeaders();
```

Comment:
Can we factor this out into a separate function? This function is getting too long, and a separate function will allow us to do early return and not have it be so highly nested (please make sure you have code coverage for all the early return paths).

Reply:
I have factored in to a function and changed the code the such a way that it also uses the logic of posting hosts.